storage/innobase/dict/dict0dd.cc

```diff
@@ -4474,11 +4473,9 @@ const char *dd_process_dd_tables_rec_and_mtr_commit(
   ulint *offsets = rec_get_offsets(rec, dd_tables->first_index(), NULL,
                                    ULINT_UNDEFINED, &heap);
 
-  const dd::Object_table &dd_object_table = dd::get_dd_table<dd::Table>();
-
   field = rec_get_nth_field(
       rec, offsets,
-      dd_object_table.field_number("FIELD_ENGINE") + DD_FIELD_OFFSET, &len);
+      4 + DD_FIELD_OFFSET, &len);
```

Comment:
Would it be difficult to initialize system tables registry? If yes, then we better keep using fixed column numbers :)

Reply:
Initialzing it is problematic, as it goes through the high level DD api, but I fixed this with adding a method to the low level table class.